### PR TITLE
Fixed axis type in ReverseInputChannels.

### PR DESCRIPTION
**Pull request by popovaan**
Root cause analysis: 
Several models fail as axis attribute does not have 'copy' attribute. The root cause is that axis type is changed in pass_rc_through_transpose() method from nd.array to int.

Solution: 
Set correct type in pass_rc_through_transpose() method.

Ticket: 71427

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests
* [x]  Framework operation tests
* [x]  Transformation tests
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list - N/A
* [x]  Guide on how to convert the **public** model - N/A
* [x]  User guide update - N/A

### CODE REVIEW

File: model-optimizer/unit_tests/extensions/back/ReverseInputChannels_test.py [hunk 1]
```diff
@@ -3,6 +3,8 @@
 
 import unittest
 
+import numpy as np
```
Comment (sadolini) [inline model-optimizer/unit_tests/extensions/back/ReverseInputChannels_test.py:6]: nit: sort import (move numpy before  unittest)

**Review by sadolini — APPROVED** [review]

**Review by evolosen — APPROVED** [review]

File: model-optimizer/unit_tests/extensions/back/ReverseInputChannels_test.py (hunk 1 above)
Comment (popovaan) [reply model-optimizer/unit_tests/extensions/back/ReverseInputChannels_test.py:6]: Already sorted. PEP8 sets numpy after unittest.

**Review by pavel-esir — APPROVED** [review]

File: model-optimizer/extensions/back/ReverseInputChannels.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ class InsertReverseChannels(BackReplacementPattern):
     enabled = False
 
     @staticmethod
-    def get_fw_index(node: Node, idx: int):
+    def get_fw_index(node: Node, idx: np.ndarray) -> np.ndarray:
```
Comment (rkazants) [inline model-optimizer/extensions/back/ReverseInputChannels.py:54]: I see no reason to change `int` -> `np.ndarray`. In all callers we pass scalar int.

File: model-optimizer/extensions/back/ReverseInputChannels.py [hunk 3]
```diff
@@ -146,7 +146,7 @@ def pass_rc_through_transpose(node: Node, reverse_channels: Node):
         if order is None:
             return False
         new_axis = list(order).index(reverse_axis)
-        reverse_channels.axis = new_axis
+        reverse_channels.axis = int64_array(new_axis)
```
Comment (rkazants) [inline model-optimizer/extensions/back/ReverseInputChannels.py:149]: May we obtain a problem with code lines like `if shape[reverse_channels.axis] == 1:`? Is reverse_channels.axis always 0D array?

Comment (popovaan) [reply model-optimizer/extensions/back/ReverseInputChannels.py:149]: In MO we set only 0D arrays as axis. There is also a transformation [UnpackPackReverseInputChannels](https://github.com/openvinotoolkit/openvino/blob/master/model-optimizer/extensions/front/tf/UnpackPackReverseInputChannels.py#L67) which inserts ReverseChannels with axis from [Stack](https://www.tensorflow.org/api_docs/python/tf/stack) and [Unstack](https://www.tensorflow.org/api_docs/python/tf/unstack) operations. Axis is int in both operartions.

But still if axis will be not a 0D tensor then pass_rc_through_transpose(), lift_up_through_transpose() will fail with errors. So I think check is needed.

File: model-optimizer/extensions/back/ReverseInputChannels.py (hunk 2 above)
Comment (popovaan) [reply model-optimizer/extensions/back/ReverseInputChannels.py]: Changed to int.

File: model-optimizer/extensions/back/ReverseInputChannels.py (hunk 3 above)
Comment (popovaan) [reply model-optimizer/extensions/back/ReverseInputChannels.py:153]: Added checks in pass_rc_through_transpose(), lift_up_through_transpose().

File: model-optimizer/extensions/back/ReverseInputChannels.py [hunk 4]
```diff
@@ -75,7 +75,7 @@ def get_fw_index(node: Node, idx: int):
 
         assert len(order) > idx >= 0, \
             'Channel index {} is incompatible with old_api_map in node {}.'.format(idx, node_name)
-        return list(order).index(idx)
+        return int64_array(list(order).index(idx))
```
Comment (rkazants) [inline model-optimizer/extensions/back/ReverseInputChannels.py:78]: it still seems to be of type numpy.ndarray

Comment (popovaan) [reply model-optimizer/extensions/back/ReverseInputChannels.py]: Fixed.

File: model-optimizer/extensions/back/ReverseInputChannels.py (hunk 3 above)
Comment (popovaan) [reply model-optimizer/extensions/back/ReverseInputChannels.py:150]: Discussed offline, removed checks as they are not needed.

**Review by rkazants — APPROVED** [review]

**Pull request merged**